contents/testing/integration-testing.md

```diff
@@ -0,0 +1,224 @@
+<frontmatter>
+  title: Integration Testing
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+{{ navbar | safe }}
+
+<div class="website-content">
+
+# Integration Testing
+
+**Author(s): [Jacob Li PengCheng](https://github.com/jacoblipech)**
+
+Reviewers: [Amrut Prabhu](https://github.com/amrut-prabhu), [Marvin Chin Kun Song](https://github.com/marvinchin), [Lin Si Jie](https://github.com/sijie123)
+
+## What is integration testing?
+
+Integration testing is a part of software testing where individual parts of your application are combined and tested as a group. This usually happens after the [<tooltip content="Testing of individual components within a system">unit testing stage</tooltip>](http://softwaretestingfundamentals.com/unit-testing/) and before the [<tooltip content="Evaluation of the software against requirements gathered from users and system specifications">system testing stage</tooltip>](https://www.guru99.com/system-testing.html) and it aims to discover faults related to the interaction between components.
+
+Imagine that you have built an online web ordering system and a sample architecture diagram of your application is shown as follows:
+
+<center>
+
+![Sample architecture diagram](integration-test/sample-architecture.png "Sample architecture diagram")
+</center>
+
+When focusing on the food ordering aspect of this system, a unit test on the `Order Summary` could look like this:
+```
+@Test
+public void isOrderSummaryCorrect(OrderSummary orderSummary) {
+	assertTrue(orderSummary.contains("cheese burger"));
+}
+```
+
+After unit testing of individual modules are completed, an integration test on `Order Food` and `Order Summary` could look like this:
+
+```
+public class FoodOrderingIntegrationTest {
+	public void testFoodOrderRecorded() {
+		FoodOrder foodOrder1 = new FoodOrder("cheese burger");
+		FoodSummary foodSummary = new FoodSummary();
+		foodSummary.add(foodOrder);
+		
+		isOrderSummaryCorrect(foodSummary);
+	}
+}
+```
+
+The system test is used to ensure that your entire application has the necessary working functionalities such as ordering, paying and generate reports. It usually involve different scenarios of using the entire system like this:
```

Comment:
I'm not sure if discussing/explaining the unit and system tests in detail here will dilute the focus of your article.

Reply:
ohh I understand where you are coming from. I shortened this part for the readers but I believe explaining a bit of other tests can make it clearer to readers when they need integration testing!